components/attestation/isecl-get-openssl-cnf.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+TOKEN_SERVICE=${TOKEN_SERVICE:-"not.set"}
+VER_SERVICE=${VER_SERVICE:-"not.set"}
+VER_SERV_USERNAME=${VER_SERV_USERNAME:-"admin"}
+VER_SERV_PASSWD=${VER_SERV_PASSWD:-"password"}
+NODEHOSTNAMEFILE=/host/hostname
+
+if [ -s ${NODEHOSTNAMEFILE} ]; then
+  NODEHOSTNAME=$(cat ${NODEHOSTNAMEFILE})
+else
+  echo "ERROR ${NODEHOSTNAMEFILE} is either missing or empty"
+  exit 1
+fi
+
+
+TEMPDIR="/tmp/tsi.$$"
+
+cleanup()
+{
+  rm -rf ${TEMPDIR}
+  #echo "Cleanup executed"
+}
+
+## create help menu:
+helpme()
+{
+  cat <<HELPMEHELPME
+
+Syntax: ${0}
+
+This script returns auto-generated openssl.cnf file that
+is using ASSET_TAG key value pairs
+
+HELPMEHELPME
+}
+
+# validate the arguments
+if [[ "$1" == "-?" || "$1" == "-h" || "$1" == "--help" ]] ; then
+  helpme
+  exit 1
+fi
+
+mkdir -p ${TEMPDIR}
+TOKEN_FILE=${TEMPDIR}/token
+
+# get a token:
+SC=$(curl -k --location --max-time 5 -s -w "%{http_code}" -o $TOKEN_FILE --request POST "${TOKEN_SERVICE}/aas/token" \
+ --header 'Content-Type: application/json' \
+ --data-raw '{ "username": "'${VER_SERV_USERNAME}'", "password": "'${VER_SERV_PASSWD}'" }')
+
+if [ "$SC" != "200" ] || [ ! -s "${TOKEN_FILE}" ]; then
+  echo "ERROR obtaining a token from Verfication Service: $SC"
+  cleanup
```

Comment:
since cleanup is consistent across every branch it may be easier to have the body be in a function and call

```
main_run
cleanup
```

Reply:
well, we want to clean up then exit with error code when error happens. Shell is not good for passing return values across functions

Reply:
yea that's fine. okay with the way it is.